### DISCUSSION TITLE
How to close sessions on Postgresql?

### DISCUSSION THREAD

**Discussion by junoriosity** [sqlalchemy/sqlalchemy#10752] [Usage Questions] [answered]
In this configuration

```
from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine, async_sessionmaker

db_engine = create_async_engine('<DATABASE_URL>', echo=False, future=True)
async_session = async_sessionmaker(db_engine, class_=AsyncSession, expire_on_commit=False)


async def get_session() -> AsyncSession:
    async with async_session() as session:
        yield session
        await session.close()
```

I can connect to the Postgresql DB, but by some several connections stay open. Is that an issue and how can I fix it?

**Comment by CaselIT** [comment, marked as answer]
Hi,

SQLAlchemy uses connection pooling, as described here: https://docs.sqlalchemy.org/en/20/core/pooling.html
You can disable it by using a null pool https://docs.sqlalchemy.org/en/20/core/pooling.html#sqlalchemy.pool.NullPool
